crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3133,35 +3097,46 @@ impl<'db> TypeInferenceBuilder<'db> {
                     self.file().path(self.db())
                 );
                 report_unresolved_module(&self.context, module_ref, *level, module);
-                self.add_unknown_declaration_with_binding(alias.into(), definition);
-                return;
+                return None;
             }
         };
 
         let Some(module_ty) = self.module_type_from_name(&module_name) else {
             report_unresolved_module(&self.context, module_ref, *level, module);
+            return None;
+        };
+
+        Some((module_name, module_ty))
+    }
+
+    fn infer_import_from_definition(
+        &mut self,
+        import_from: &'db ast::StmtImportFrom,
+        alias: &ast::Alias,
+        definition: Definition<'db>,
+    ) {
+        let Some((module_name, module_ty)) = self.resolve_import_from_module(import_from, alias)
+        else {
             self.add_unknown_declaration_with_binding(alias.into(), definition);
             return;
         };
 
-        let ast::Alias {
-            range: _,
-            name,
-            asname: _,
-        } = alias;
+        let star_import_info =
+            if let DefinitionKind::StarImport(star_import) = definition.kind(self.db()) {
+                Some((star_import, symbol_table(self.db(), self.scope())))
```

Comment:
```suggestion
                Some((star_import, self.index.symbol_table(self.scope())))
```

Reply:
hmm, that doesn't compile. This does compile:

```suggestion
                Some((
                    star_import,
                    self.index
                        .symbol_table(self.scope().file_scope_id(self.db())),
                ))
```

but it doesn't really seem simpler anymore ;-)

Reply:
Oh I see — I'll make the change, thanks